sechub-wrapper-owasp-zap/src/main/java/com/mercedesbenz/sechub/owaspzapwrapper/util/TargetConnectionChecker.java

```diff
@@ -0,0 +1,122 @@
+        }
+        if ("https".equals(urlToCheckConnection.getProtocol())) {
+            TrustManager[] trustAllCerts = createTrustAllCertsManager();
+
+            installAllTrustingTrustManager(urlToCheckConnection, trustAllCerts);
+
+            HostnameVerifier allHostsValid = createAllTrustingHostNameVerifier();
+            // Install the all-trusting host verifier
+            HttpsURLConnection.setDefaultHostnameVerifier(allHostsValid);
+        }
+
+        try {
+            LOG.info("Trying to reach target URL: {}", urlToCheckConnection.toExternalForm());
+            HttpURLConnection connection;
+            if (proxyInformation == null) {
+                connection = (HttpURLConnection) urlToCheckConnection.openConnection();
+            } else {
+                Proxy proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(proxyInformation.getHost(), proxyInformation.getPort()));
+                connection = (HttpURLConnection) urlToCheckConnection.openConnection(proxy);
+            }
+            int responseCode = connection.getResponseCode();
+            if (isReponseCodeValid(responseCode)) {
+                LOG.info("Target is reachable.");
+                return true;
+            } else {
+                LOG.info("Target is NOT reachable. Aborting Scan...");
+            }
+        } catch (IOException e) {
+            LOG.error("An exception occurred while checking if target URL is reachable: {} because: {}", urlToCheckConnection.toExternalForm(), e.getMessage());
+        }
+        return false;
+    }
+
+    static boolean isReponseCodeValid(int responseCode) {
+        return responseCode < 400 || responseCode == 401 || responseCode == 403;
+    }
+
+    private static TrustManager[] createTrustAllCertsManager() {
+        TrustManager[] trustAllCerts = new TrustManager[] { new X509TrustManager() {
+            public X509Certificate[] getAcceptedIssuers() {
+                return null;
+            }
+
+            public void checkClientTrusted(X509Certificate[] certs, String authType) {
+            }
+
+            public void checkServerTrusted(X509Certificate[] certs, String authType) {
+            }
+        } };
+        return trustAllCerts;
+    }
+
+    private static void installAllTrustingTrustManager(URL targetUrl, TrustManager[] trustAllCerts) {
+        try {
+            SSLContext sslContext = SSLContext.getInstance("SSL");
+            sslContext.init(null, trustAllCerts, new SecureRandom());
+            HttpsURLConnection.setDefaultSSLSocketFactory(sslContext.getSocketFactory());
+        } catch (KeyManagementException | NoSuchAlgorithmException e) {
+            LOG.error("An exception occurred while installing the all-trusting trust manager for checking if target URL is reachable: {}",
+                    targetUrl.toExternalForm(), e);
+        }
+    }
+
+    private static HostnameVerifier createAllTrustingHostNameVerifier() {
```

Comment:
```suggestion
    private static HostNameVerifier createAllTrustingHostNameVerifier() {
```

Reply:
This is an interface `javax.net.ssl.HostnameVerifier` that was just used here.